plugins/samson_ledger/config/initializers/samson_ledger/client.rb

```diff
@@ -0,0 +1,65 @@
+# frozen_string_literal: true
+#
+# basic ledger event client
+#
+require 'faraday'
+require 'openssl'
+
+module Ledger
+  class Client
+    LEDGER_URI = "/api/v1/events"
+    class << self
+      def plugin_enabled?
+        ENV["LEDGER_TOKEN"] && ENV["LEDGER_BASE_URL"]
+      end
+
+      def post_deployment(deploy)
+        verify_env
+        post_event(deploy)
+      end
+
+      private
+
+      def verify_env
+        raise "LEDGER_TOKEN is not set in ENV" unless ENV.fetch("LEDGER_TOKEN")
+        raise "LEDGER_BASE_URL is not set in ENV" unless ENV.fetch("LEDGER_BASE_URL")
```

Comment:
can simply fetch when used inside the other code since this should never happen because the plugin_enabled guards against it